core/kazoo_number_manager/src/knm_services.erl

```diff
@@ -148,7 +148,8 @@ update_services(Number, 'false', _) ->
     _ = kz_services:reconcile(PrevAssignedTo, <<"phone_numbers">>),
     Services = fetch_services(Number),
     Transactions = knm_number:transactions(Number),
-    _ = kz_services:commit_transactions(Services, Transactions),
+    _ = 'undefined' =/= AssignedTo
```

Comment:
How about checking this earlier? Do we want to compute services on unassigned numbers?